content/realtime/authentication.textile

```diff
@@ -252,12 +257,23 @@ h4. Parameters
 - <div lang="ruby">&block</div> := yields a "@TokenDetails@":#token-details object
 - <div lang="swift,objc">callback</div> := called with a "@ARTTokenDetails@":#token-details object or an error
 
-blang[jsall,objc,swift].
+blang[objc,swift].
   h4. Callback result
 
   On success, @tokenDetails@ contains a "@TokenDetails@":#token-details object containing the details of the new or existing token along with the token string.
 
-  On failure to obtain a token, @err@ contains an <span lang="ruby">"@ErrorInfo@":#error-info</span><span lang="objc,swift">@NSError@</span> object with an error response as defined in the "Ably REST API":/rest-api#common documentation.
+  On failure to obtain a token, @err@ contains an @NSError@ object with an error response as defined in the "Ably REST API":/rest-api#common documentation.
+
+blang[jsall].
+  h4. Callback result
+
+  If using the REST client, the callback will be called once the call to obtain a new token returns, with either @tokenDetails@ as a "@TokenDetails@":#token-details object if the call succeeded, or @err@ as an "@ErrorInfo@":#error-info (with an error response as defined in the "Ably REST API":/rest-api#common documentation) if it failed.
+
+  If using the realtime client, if the call to obtain a new token failed, the callback will be called with @err@ as with the REST client. If it succeeded, the callback will be called (with the new @tokenDetails@) only once the realtime connection has been successfully upgraded to use the new token (or if the upgrade failed, once the connection state has moved to @FAILED@).
+
+  If the client was not previously connected, the callback will be called with @tokenDetails@ once it has successfully connected with the new token (or closed, failed, or suspended, with an @err@, should that happen first).
```

Comment:
I can't recall why we decided on doing the auto-connect.  It does seem a little odd now when reading the docs, although I can equally see almost no situation where someone using the Ably Realtime lib would not want to connect.